src/www/ui/api/Controllers/CopyrightController.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/*
+ Author: Soham Banerjee <sohambanerjee4abc@hotmail.com>
+ SPDX-FileCopyrightText: © 2023 Soham Banerjee <sohambanerjee4abc@hotmail.com>
+
+ SPDX-License-Identifier: GPL-2.0-only
+*/
+
+/**
+ * @file
+ * @brief Controller for copyright queries
+ */
+
+namespace Fossology\UI\Api\Controllers;
+
+use Fossology\Lib\Auth\Auth;
+use Fossology\Lib\Dao\UploadDao;
+use Fossology\Lib\Db\DbManager;
+use Fossology\UI\Api\Helper\ResponseHelper;
+use Fossology\UI\Api\Models\Info;
+use Fossology\UI\Api\Models\InfoType;
+use Psr\Http\Message\ServerRequestInterface;
+
+
+class CopyrightController extends RestController
+{
+  /**
+   * @var ContainerInterface $container
+   * Slim container
+   */
+  protected $container;
+
+  /**
+   * Get query parameter name for copyright filtering
+   */
+  const COPYRIGHT_PARAM = "status";
+
+  /**
+   * @var copyrightHist $copyrightHist
+   * Copyright Histogram object
+   */
+  private $copyrightHist;
+
+  public function __construct($container)
+  {
+    parent::__construct($container);
+    $this->restHelper = $container->get('helper.restHelper');
+    $this->copyrightHist = $this->restHelper->getPlugin('ajax-copyright-hist');
+  }
+
+  /**
+   * Get all copyrights for a particular upload-tree
+   *
+   * @param  ServerRequestInterface $request
+   * @param  ResponseHelper         $response
+   * @param  array                  $args
+   * @return ResponseHelper
+   */
+  public function getFileCopyrights($request, $response, $args)
+  {
+    $uploadPk = $args["id"];
+    $uploadTreeId = $args["itemId"];
+    $query = $request->getQueryParams();
+    $statusVal = true;
+    $returnVal = null;
+    if (!array_key_exists(self::COPYRIGHT_PARAM, $query)) {
+      $returnVal = new Info(400, "Bad Request. 'status' is a required query param", InfoType::ERROR);
```

Comment:
Would be better to send expected values as `active` and `inactive` as well.

Reply:
You mean in the error message right?

Reply:
Yes